src/core/index.ts

```diff
@@ -40,39 +40,48 @@ function resultContainer<TValue>(): ResultContainer<TValue> {
   }
 }
 
-const createRenderHook = <TProps, TResult, TOptions extends {}, TRenderer extends Renderer<TProps>>(
+function createRenderHook<TProps, TResult, TOptions extends {}, TRenderer extends Renderer<TProps>>(
   createRenderer: CreateRenderer<TProps, TResult, TOptions, TRenderer>
-) => (
-  callback: (props: TProps) => TResult,
-  options: RenderHookOptions<TProps, TOptions> = {} as RenderHookOptions<TProps, TOptions>
-): RenderHook<TProps, TResult, TRenderer> => {
-  const { result, setValue, setError, addResolver } = resultContainer<TResult>()
-  const renderProps = { callback, setValue, setError }
-  let hookProps = options.initialProps
+) {
+  const renderHook = (
+    callback: (props: TProps) => TResult,
+    options: RenderHookOptions<TProps, TOptions> = {} as RenderHookOptions<TProps, TOptions>
+  ): RenderHook<TProps, TResult, TRenderer> => {
+    const { result, setValue, setError, addResolver } = resultContainer<TResult>()
+    const renderProps = { callback, setValue, setError }
+    let hookProps = options.initialProps
 
-  const { render, rerender, unmount, act, ...renderUtils } = createRenderer(renderProps, options)
+    const { render, rerender, unmount, act, ...renderUtils } = createRenderer(renderProps, options)
 
-  render(hookProps)
+    render(hookProps)
 
-  function rerenderHook(newProps = hookProps) {
-    hookProps = newProps
-    rerender(hookProps)
-  }
+    const rerenderHook = (newProps = hookProps) => {
+      hookProps = newProps
+      rerender(hookProps)
+    }
 
-  function unmountHook() {
-    removeCleanup(unmountHook)
-    unmount()
-  }
+    const unmountHook = () => {
+      removeCleanup(unmountHook)
+      unmount()
+    }
 
-  addCleanup(unmountHook)
+    addCleanup(unmountHook)
 
-  return {
-    result,
-    rerender: rerenderHook,
-    unmount: unmountHook,
-    ...asyncUtils(act, addResolver),
-    ...renderUtils
+    return {
+      result,
+      rerender: rerenderHook,
+      unmount: unmountHook,
+      ...asyncUtils(act, addResolver),
+      ...renderUtils
+    }
   }
+
+  // If the function name does not get used before it is returned,
+  // it seems to vanish in nodejs and does not appear in stack traces.
+  // This dummy usage works around that.
+  const _name = renderHook.name // eslint-disable-line @typescript-eslint/no-unused-vars
```

Comment:
This is gross and I have no idea why it is necessary, but in node, if I've got function that returns a function like so:
```js
function createFunc() {
  function func() {
  }
  return func
}
```

then:

```js
console.log(createFunc.name) // 'createFunc'
console.log(createFunc().name) // ''
```

But if I use `func.name` inside `createFunc` like so:

```js
function createFunc() {
  function func() {
  }
  const name = func.name // not returned, just used
  return func
}
```

then:

```js
console.log(createFunc.name) // 'createFunc'
console.log(createFunc().name) // 'func'
```

In the chrome console, both example will return `func` for the second log.  It's very weird.

Reply:
That is really weird. Can't seem to find much online either how to potentially solve.

Reply:
So I've discovered that this isn't a node thing, but something in the compilation.  I've narrowed it down to babel (not TS and not something in `jest`) as if I use `kcd-scripts` to compile a javascript only file, this

```js
export function createFunc1() {
  function func() {
  }
  return func
}

export function createFunc2() {
  function func() {
  }
  const name = func.name // not returned, just used
  return func
}
```

becomes 

```js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createFunc1 = createFunc1;
exports.createFunc2 = createFunc2;

function createFunc1() {
  return function () {};
}

function createFunc2() {
  function func() {}

  func.name; // not returned, just used

  return func;
}
```

The name in `createFunc1` has been scrubbed!

Now I just have to work out which plugin is causing this.

Reply:
its babel-plugin-minify-dead-code-elimination

https://babeljs.io/en/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=usage&spec=false&loose=false&code_lz=GYVwdgxgLglg9mABBATgUwIZTQMXBACgEpEBvAKEUVElgWv2LMsQF8X0oQUkaJz25CAgDOcADZoAdOLgBzAqkzY8kKWAwBbNESGiJ02QqVZcjIuq06gA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=true&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ces2015%2Creact%2Ctypescript&prettier=true&targets=&version=7.12.12&externalPlugins=babel-plugin-minify-dead-code-elimination%400.5.1

Reply:
Thanks @jpeyper.  I've raised [an issue in `kcd-scripts`](https://github.com/kentcdodds/kcd-scripts/issues/188) about this.

Reply:
I'll keep an eye on the other issue, in the meantime, we have this workaround 🤷